dependencies.yaml

```diff
@@ -41,23 +41,28 @@ dependencies:
             packages:
               - cudatoolkit=11.2
               - gcc<11.0.0
+              - sysroot_linux-64==2.17
```

Comment:
Just after we switched over to dependencies.yaml we ran into link issues with some of the tests that called cuda runtime funcitons. I will roll back these changes and see if other CI updates have made this unneeded

Reply:
Removing these lines results in CI failures for `11.4.1, ubuntu20.04, amd64, 3.8, v100, 520` and `11.8.0, ubuntu22.04, amd64, 3.10, v100, 520`, so it looks like we need to pin glibc version so that we have the correct symbols for building the `rapids_test_generate_resource_spec` executable.

```
-- rapids_test_generate_resource_spec compile failure details are /opt/conda/envs/test/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /opt/conda/envs/test/lib/libcudart.so: undefined reference to `memcpy@GLIBC_2.14'
/opt/conda/envs/test/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: /opt/conda/envs/test/lib/libcudart.so: undefined reference to `clock_gettime@GLIBC_2.17'
```